www/src/pages/table/OntSearchComponent.tsx

```diff
@@ -41,59 +46,132 @@ function useSearchComponent(
+            });
+          });
+          return searchResults;
+        })
+        .then((searchResults) => {
+          setSearchOptions([...searchResults]);
+        })
+        .catch(function (error: any) {
+          console.error(error);
+        });
+    } else if (storeName === "propertyStore") {
+      store
+        .fetchSearchResults(query)
+        .then((data) => {
+          data.forEach((searchResult: PropertyTextSearchResult) => {
+            searchResults.push({
+              id: searchResult.id,
+              label: (
+                <LabelComponent
+                  id={searchResult.id}
+                  label={searchResult.label}
+                  description={searchResult.description}
+                />
+              ),
+              filterlabel: `${searchResult.label} (${searchResult.id})`,
+              value: `${searchResult.id}`,
+            });
+          });
+          return searchResults;
+        })
+        .then((searchResults) => {
+          setSearchOptions([...searchResults]);
+        })
+        .catch(function (error: any) {
+          console.error(error);
+        });
+    } else {
+      store
+        .fetchSearchResults(query)
+        .then((data) => {
+          data.forEach((searchResult: EntityTextSearchResult) => {
+            searchResults.push({
+              id: searchResult.id,
+              label: (
+                <LabelComponent
+                  id={searchResult.id}
+                  label={searchResult.label}
+                  description={searchResult.description}
+                />
+              ),
+              filterlabel: `${searchResult.label} (${searchResult.id})`,
+              value: `${searchResult.id}`,
+            });
+          });
+          return searchResults;
+        })
+        .then((searchResults) => {
+          setSearchOptions([...searchResults]);
+        })
+        .catch(function (error: any) {
+          console.error(error);
+        });
     }
-    return options;
-  }, [store.records.size]);
-
-  // search for additional properties if it's not in the list
-  const onSearch = (query: string) => {
-    if (query === "") return;
-    seqInvoker.exec(() => {
-      return store.fetchById(query).then(() => 1);
-    });
+    console.log(searchResults);
   };
 
   return (
     <Select
-      mode={props.mode}
       allowClear={true}
-      options={options}
-      optionFilterProp="label"
+      options={searchOptions}
+      onClear={() => setSearchOptions([])}
+      optionFilterProp="filterlabel"
+      defaultActiveFirstOption={false}
       className={props.classes.selection}
       showSearch={true}
-      onSearch={onSearch}
-      value={props.value}
-      onSelect={props.onSelect as any}
-      onDeselect={props.onDeselect as any}
+      onSearch={debounce(onSearch, 300)}
+      value={props.value === undefined ? undefined : `${props.value}`}
+      onSelect={(value: any, option: SearchOptions) => {
```

Comment:
let's check if`props.onSelect` is undefined before creating a function

Reply:
fixed

Reply:
not like this. It should be `onSelect={props.onSelect === undefined ? undefined : (value: any, ...)}`, also you can remove `?` in `props.onSelect?.`